.github/workflows/lychee.yml

```diff
@@ -0,0 +1,48 @@
+# This runs `cargo lychee` on all directories listed.
+# <https://github.com/lycheeverse/lychee>
+
+name: Lychee
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+  workflow_dispatch:
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  audit:
+
+    runs-on: ubuntu-latest
+
+    steps:
+
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        submodules: recursive
+
+    - name: Cache
+      uses: actions/cache@v4
+      with:
+        path: |
+          ~/.cargo
+          target
+        key: lychee
+
+    - name: Install dependencies
+      run: cargo install lychee --locked
+
+    - name: Documentation
+      run: cargo doc --workspace --all-features --no-deps
+
+    # Uses `lychee.toml` at:
+    # 1. The repo root (excluding `target/`)
+    # 2. Cuprate's cargo documentation
+    - name: Lychee
+      run: |
+        lychee . --exclude target
+        lychee target/doc/cuprate_*
```

Comment:
`cargo doc` is ran before such that we can do `lychee target/doc/cuprate_*` and scan all of the generated docs.

This is for stuff like `rpc/types` where the links are generated and are not in the source.